athenian/database_pg.go

```diff
@@ -223,6 +224,85 @@ func (db *database) AthenianToGithub(ctx context.Context, id AccountID) ([]Githu
 	return out, rows.Err()
 }
 
+func (db *database) GetInstaflowStatus(ctx context.Context, accID AccountID) (*InstaflowStatus, error) {
+	return db.getInstaflowStatus(ctx, accID)
+}
+
+func (db *database) getInstaflowStatus(ctx context.Context, accID AccountID) (*InstaflowStatus, error) {
+	row := db.db.QueryRow(
+		ctx,
+		`SELECT account_id, account_created, fetch_started, fetch_completed, consistency_started, consistency_completed, precompute_started, precompute_completed, current_status
+		FROM public.installation_progress WHERE account_id = $1`,
+		int64(accID),
+	)
+	instaflowStatus, err := scanInstaflowStatus(row)
+	if err == pgx.ErrNoRows {
+		err = ErrNotFound
+	}
+	return &instaflowStatus, err
+}
+
+func scanInstaflowStatus(sc dbs.Scanner) (InstaflowStatus, error) {
+	var (
+		accID                int64
+		accountCreated       *time.Time
+		fetchStarted         *time.Time
+		fetchCompleted       *time.Time
+		consistencyStarted   *time.Time
+		consistencyCompleted *time.Time
+		precomputeStarted    *time.Time
+		precomputeCompleted  *time.Time
+		status               string
+	)
+	err := sc.Scan(&accID, &accountCreated, &fetchStarted, &fetchCompleted, &consistencyStarted, &consistencyCompleted, &precomputeStarted, &precomputeCompleted, &status)
+	if err == pgx.ErrNoRows {
+		err = ErrNotFound
+	}
+
+	return InstaflowStatus{
+		AccID:                AccountID(accID),
+		AccountCreated:       accountCreated.UTC(),
+		FetchStarted:         fetchStarted.UTC(),
+		FetchCompleted:       fetchCompleted.UTC(),
+		ConsistencyStarted:   consistencyStarted.UTC(),
+		ConsistencyCompleted: consistencyCompleted.UTC(),
+		PrecomputeStarted:    precomputeStarted.UTC(),
+		PrecomputeCompleted:  precomputeCompleted.UTC(),
+		Status:               InstallStatus(status),
+	}, err
+}
+
+func (db *database) UpdateInstaflowStatus(ctx context.Context, accID AccountID, timestamp time.Time, status InstallStatus) error {
+	tx, err := db.db.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(ctx)
+
+	var qu string
+
+	ext, err := db.getInstaflowStatus(ctx, accID)
+	if err == ErrNotFound {
+		qu = fmt.Sprintf(`INSERT INTO public.installation_progress (account_id, %s, current_status)
+		VALUES ($1, $2, $3)
+		ON CONFLICT (account_id) DO UPDATE SET %s = $2, current_status = $3 where account_id = $1`, status, status)
```

Comment:
hmm I was thinking that the source for statuses is only handle event functions. and it is uses enum directly.
but for the future yes we can add some specific hints to avoid if the status is not in this enum.
updated